R/result.R

```diff
@@ -163,124 +172,109 @@ BenchmarkResult <- R6Point1Class(
+        result_df,
+        dplyr::as_tibble(x$optional_benchmark_info$params)
       )
-      out$output <- x$output
+      out$output <- x$optional_benchmark_info$output
 
-      # append metadata fields to dataframe as attributes
-      metadata_elements <- c("name", "tags", "info", "context", "github", "options")
+      # append other fields to dataframe as attributes
+      metadata_elements <- setdiff(names(x), "optional_benchmark_info")
       for (element in metadata_elements) {
-        if (element %in% names(x)) {
-          attr(out, element) <- x[[element]]
-        }
+        attr(out, element) <- x[[element]]
       }
 
       out
     }
   ),
 
   active = list(
-    name = function(name) private$get_or_set_serializable(variable = "name", value = name),
-    result = function(result) private$get_or_set_serializable(variable = "result", value = result),
-    params = function(params) private$get_or_set_serializable(variable = "params", value = params),
+    run_name = function(run_name) private$get_or_set_serializable(variable = "run_name", value = run_name),
+    run_id = function(run_id) private$get_or_set_serializable(variable = "run_id", value = run_id),
+    batch_id = function(batch_id) private$get_or_set_serializable(variable = "batch_id", value = batch_id),
+    run_reason = function(run_reason) private$get_or_set_serializable(variable = "run_reason", value = run_reason),
+    timestamp = function(timestamp) private$get_or_set_serializable(variable = "timestamp", value = timestamp),
+    stats = function(stats) private$get_or_set_serializable(variable = "stats", value = stats),
+    error = function(error) private$get_or_set_serializable(variable = "error", value = error),
+    validation = function(validation) private$get_or_set_serializable(variable = "validation", value = validation),
     tags = function(tags) private$get_or_set_serializable(variable = "tags", value = tags),
     info = function(info) private$get_or_set_serializable(variable = "info", value = info),
+    optional_benchmark_info = function(optional_benchmark_info) private$get_or_set_serializable(variable = "optional_benchmark_info", value = optional_benchmark_info),
+    machine_info = function(machine_info) private$get_or_set_serializable(variable = "machine_info", value = machine_info),
+    cluster_info = function(cluster_info) private$get_or_set_serializable(variable = "cluster_info", value = cluster_info),
     context = function(context) private$get_or_set_serializable(variable = "context", value = context),
     github = function(github) private$get_or_set_serializable(variable = "github", value = github),
-    options = function(options) private$get_or_set_serializable(variable = "options", value = options),
-    output = function(output) private$get_or_set_serializable(variable = "output", value = output),
-    rscript = function(rscript) private$get_or_set_serializable(variable = "rscript", value = rscript),
```

Comment:
I know that we can't send it yet, but it feels a little silly to pull this out when we will then put it back in (ideally) very soon. Anyway that we can not delete this just to re-add it later?

Reply:
Per one of our earlier conversations, they're removed from the top-level fields here so the keys exactly correspond to the `/api/benchmarks/` endpoint, but we're not losing data, because we're sticking it all in `optional_benchmark_info` starting here: https://github.com/voltrondata-labs/arrowbench/pull/116/files/1019e29cc2053afc143943b6bf5094c50af0e903#diff-c118b736cd58f32580f07445199c10b7fec49b44f25f27cd46bbf1215cc9eff3R241-R247 (and `output` and `rscript` are appended later). They're not part of the schema anymore, though.

For a bit I tried to make a slim result object that was just the API schema and a richer one that had extra metadata like this that inherited from it, but it got pretty complicated and ugly, to be honest. This way is less structured, but what structure we have is all consistent such that we don't have 6 versions of results schemas like before, which is nice.

Reply:
Aaah right, sticking them in `optional_benchmark_info` is great! I didn't see that directly here, but now see it  in `run.R` Thanks!